packages/core/src/lib/markdown-it/highlight/HighlightRule.ts

```diff
@@ -61,56 +66,33 @@ export class HighlightRule {
     return atLineNumber;
   }
 
-  applyHighlight(line: string, lineNumber: number) {
-    // Applied rule is the first component until deduced otherwise
+  getHighlightType(lineNumber: number): {
+    highlightType: HIGHLIGHT_TYPES,
+    bounds: Array<[number, number]> | null
+  } {
     let [appliedRule] = this.ruleComponents;
-
     if (this.isLineRange()) {
-      // For cases like 2[:]-3 (or 2-3[:]), the highlight would be line highlight
-      // across all the ranges
-      const shouldWholeLine = this.ruleComponents.some(comp => comp.isUnboundedSlice());
-      if (shouldWholeLine) {
-        return HighlightRule._highlightWholeLine(line);
+      if (this.ruleComponents.some(comp => comp.isUnboundedSlice())) {
+        return { highlightType: HIGHLIGHT_TYPES.WholeLine, bounds: null };
       }
 
       const [startCompare, endCompare] = this.ruleComponents.map(comp => comp.compareLine(lineNumber));
       if (startCompare < 0 && endCompare > 0) {
         // In-between range
-        return HighlightRule._highlightWholeText(line);
+        return { highlightType: HIGHLIGHT_TYPES.WholeText, bounds: null };
       }
 
-      // At the range boundaries
```

Comment:
Same as above, also would be great if we have an example in the comment for each case

Reply:
done